channels/smartcard/client/smartcard_operations.c

```diff
@@ -547,9 +547,6 @@ static LONG smartcard_ListReadersA_Call(SMARTCARD_DEVICE* smartcard, SMARTCARD_O
 	cchReaders = SCARD_AUTOALLOCATE;
 	status = ret.ReturnCode = SCardListReadersA(operation->hContext, (LPCSTR) call->mszGroups,
 	                          (LPSTR) &mszReaders, &cchReaders);
-	cchReaders = filter_device_by_name_a(smartcard->names, &mszReaders, cchReaders);
```

Comment:
why move this below?

Reply:
I problem I'm seeing is that SCardListReadersA() is returning an error and it may leave mszReaders unchanged, so there's nothing to filter and (before the other part of my pull request) filter_device_by_name_a() would try to dereference that NULL value by passing it to strnlen().  With the change to filter_device_by_name_a(), it's safe to pass it a NULL, but trying to filter a list before you've determined whether that list was successfully retrieved is just asking for trouble in my opinion.